Functions/xASL_quant_M0.m

```diff
@@ -133,39 +133,33 @@
     %% ------------------------------------------------------------------------------------------------------
     % 6. Quantify the M0, either for single 3D volume or slice-wise
     if strcmpi(x.readout_dim,'3D') % for 3D readout, we assume the M0 readout is at the end of the TR
-            NetTR = TR;
-            fprintf('%s\n','Single 3D M0 readout assumed');
+		NetTR = TR;
+		fprintf('%s\n','Single 3D M0 readout assumed');
     elseif  strcmpi(x.readout_dim,'2D') % for 2D readouts, there are slice timing differences
 
-            % Calculate SliceReadoutTime for shortestTR
-            [x] = xASL_quant_SliceReadoutTime_ShortestTR(x);
-
-            SliceIM = zeros(size(M0IM));
-            for iZ=1:size(M0IM,3)
-                SliceIM(:,:,iZ) = iZ;
-            end
-
-            if  x.M0_usesASLtiming 
-				% in this case, the M0 readout has the exact same timing as the ASL readout
-				% this is the case e.g. for Philips 3D GRASE
-				if length(x.Q.SliceReadoutTime) == 1
-					NetTR = x.Q.LabelingDuration+x.Q.Initial_PLD+x.Q.SliceReadoutTime.*(SliceIM-1);
-				else
-					NetTR = x.Q.LabelingDuration+x.Q.Initial_PLD+x.Q.SliceReadoutTime(SliceIM);
-				end
-                fprintf('%s\n','2D sliceWise M0 readout assumed, same timing as ASL slices readout used');
-            elseif ~x.M0_usesASLtiming
-				if length(x.Q.SliceReadoutTime) == 1
-					NetTR = TR - ((max(SliceIM(:))-SliceIM).*x.Q.SliceReadoutTime);
-				else
-					NetTR = TR + x.Q.SliceReadoutTime(SliceIM) - max(x.Q.SliceReadoutTime) - (max(x.Q.SliceReadoutTime)-min(x.Q.SliceReadoutTime))/(size(M0IM,3)-1);
-				end
-                % Here we assume the M0 slices readout were at the end of the
-                % TR, with the same time between slices as for the ASL readout
-                fprintf('%s\n','2D SliceWise M0 readout, assumed M0 slices readout at end of TR');
-            else
-                error('Unknown x.M0_usesASLtiming specified');
-            end
+		% Calculate SliceTime as a vector
+		SliceTime = xASL_quant_SliceTimeVector(x,size(M0IM,3));
+
+		SliceIM = zeros(size(M0IM));
+		for iZ=1:size(M0IM,3)
+			SliceIM(:,:,iZ) = iZ;
+		end
+
+		if  x.M0_usesASLtiming
+			% in this case, the M0 readout has the exact same timing as the ASL readout
+			% this is the case e.g. for Philips 3D GRASE
+			NetTR = x.Q.LabelingDuration+x.Q.Initial_PLD+SliceTime(SliceIM);

```

Comment:
Is the "new" SliceTime always a vector that starts with 0?
This should be checked for all options in all *_quant_Q functions

Reply:
Yes. It is. That's what the new function does. Checks it for all cases. No need to check it afterwards, because the new function does exactly that - so you can then peacefully use it and know that it was checked already.